src/components/App/App.js

```diff
@@ -25,23 +25,109 @@ function App() {
     React.useState(false);
   const [isRegisterSuccess, setRegisterSuccess] = React.useState(false);
   const [menuOpen, setMenuOpen] = React.useState(false);
+  const [isRegistered, setIsRegistered] = React.useState(false);
+  const [token, setToken] = React.useState(localStorage.getItem("jwt"));
+  const [savedArticles, setSavedArticles] = React.useState([]);
+  const [recentArticles, setRecentArticles] = React.useState([]);
+
+  React.useEffect(() => {
+    const jwt = localStorage.getItem("jwt");
+    if (jwt) {
+      auth.checkToken(jwt).then((res) => {
+        if (res) {
+          setLoggedIn(true);
+        } else {
+          localStorage.removeItem("jwt");
+        }
+      });
+    }
+  });
+
+  React.useEffect(() => {
+    if (isLoggedIn) {
+      Promise.all([MainApi.getCurrentUser(token), MainApi.getArticles()])
+        .then(([user, articles]) => {
+          setCurrentUser(user.data);
+          setSavedArticles(articles.data);
+        })
+        .catch((err) => console.log(err));
+    }
+  }, [token, isLoggedIn]);
+
+  function onRegisterSubmit(email, password, name) {
+    auth.register(email, password, name).then((res) => {
+      if (res) {
+        setIsRegistered(true);
+        setRegisterSuccessPopup(true);
+        setFormPopup(false);
+        setRegisterSuccess(true);
+      } else {
+        console.log("ERROR");
+      }
+    });
+  }
+
+  function onLoginSubmit(email, password) {
+    auth.login(email, password).then((data) => {
+      if (data.token) {
+        localStorage.setItem("jwt", data.token);
+
+        setLoggedIn(true);
+        setFormPopup(false);
+        setToken(data.token);
+      }
+    });
+  }
 
   function registerSuccess() {
     setRegisterPopup(false);
     setRegisterSuccessPopup(false);
     setFormPopup(true);
   }
 
-  // function registerFail() {
-  //   setRegisterPopup(true);
-  //   setRegisterSuccessPopup(false);
-  //   setFormPopup(true);
-  // }
-
   function searchHandler(keyword) {
     return newsApi.searchArticles(keyword);
   }
 
+  function addArticleHandler(article) {
+    if (article) {
+      return MainApi.saveArticle(article, currentUser)
+        .then(() => {
+          setRecentArticles(savedArticles);
+        })
+        .then(() => {
+          MainApi.getArticles().then((articles) => {
+            setSavedArticles(articles.data);
+          });
+        });
+    } else {
+      throw new Error("No article added");
```

Comment:
`      throw new Error("No article added");
`


Please, don't use `throw new Error` without catching errors at the end of the request.